packages/scheduler/src/SchedulerMinHeap.js

```diff
@@ -16,73 +16,59 @@ type Node = {|
 export function push(heap: Heap, node: Node): void {
   const index = heap.length;
   heap.push(node);
-  siftUp(heap, node, index);
+  siftUp(heap, index);
 }
 
 export function peek(heap: Heap): Node | null {
-  const first = heap[0];
-  return first === undefined ? null : first;
+  if (heap.length === 0) return null;
+  return heap[0];
```

Comment:
As previous code states, heap is possibly undefined and your code tries to read length on it what may lead to a runtime error.

Reply:
Ah, my mistake, your edit has the same outcome indeed.

Reply:
I was thinking about something else in fact... Your code will fail if the first element of heap is undefined. Remember, arrays can have empty elements. So you could write something like 
`if (typeof heap[0] === 'undefined') return null;`.

Reply:
You're right but we operate the heap by `pop` and `push`, what you worry about will not happen.